src/Microsoft.AspNetCore.Mvc.Core/Internal/ActionSelector.cs

```diff
@@ -113,6 +101,37 @@ protected virtual IReadOnlyList<ActionDescriptor> SelectBestActions(IReadOnlyLis
             return actions;
         }
 
+        private IReadOnlyList<ActionDescriptor> EvaluateActionConstraints(
```

Comment:
Rename to method to something else or rename the other method to something like `EvaluateActionConstraintsCore`. This looks kind of confusing.